src/u2f/u2f_app.c

```diff
@@ -139,6 +161,16 @@ static void _app_string(const uint8_t* app_id, char* out, size_t out_len)
             return;
         }
     }
+    char* mnemonic=NULL;
+    //fallback to hex if something goes wrong
+    if (bip39_mnemonic_from_bytes(NULL, app_id, 32, &mnemonic) == WALLY_OK) {
+        const char* words[SHORT_MNEMONIC_LENGTH];
+        //second fallback
+        if(short_mnemonic(mnemonic, words)>=4) {
+            snprintf(out, out_len, "Unknown site:\n%s %s\n%s %s", words[0], words[1],words[2],words[3]);
+            return;
+        }
+    }
     char appid_hex[32 * 2 + 1] = {0};
```

Comment:
hmm I think this code is not reachable, right? 32 bytes to mnemonic should always succeed, and so should the wowrd splitting. 

Could replace it with `Abort("u2f app string")`, which halts the device in unexpected situations.